packages/fx-core/src/common/azure-hosting/utils.ts

```diff
@@ -2,23 +2,103 @@
+            .map((template) => template.Provision?.Modules)
+            .reduce((result, current) => Object.assign(result, current), {}),
+        }
+      : undefined,
     Configuration: {
       Orchestration: templates.map((template) => template.Configuration?.Orchestration).join(""),
       Modules: templates
         .map((template) => template.Configuration?.Modules)
         .reduce((result, current) => Object.assign(result, current), {}),
     },
-    Parameters: Object.assign({}, ...templates.map((template) => template.Parameters)),
+    Parameters: existsParameters
+      ? Object.assign({}, ...templates.map((template) => template.Parameters))
+      : undefined,
     Reference: Object.assign({}, ...templates.map((template) => template.Reference)),
   };
-  return result;
+}
+
+async function getAzureAccountCredential(
+  tokenProvider: TokenProvider
+): Promise<TokenCredentialsBase> {
+  const credential = await tokenProvider.azureAccountProvider.getAccountCredentialAsync();
+  if (!credential) {
+    throw new PreconditionError(Messages.FailToGetAzureCreds, [Messages.TryLoginAzure]);
+  }
+  return credential;
+}
+
+async function fetchWebSiteManagementClient(
+  tokenProvider: TokenProvider,
+  inputs: Inputs
+): Promise<appService.WebSiteManagementClient> {
+  return new appService.WebSiteManagementClient(
+    await getAzureAccountCredential(tokenProvider),
+    inputs.subscriptionId
+  );
+}
+
+async function getAzureDeployConfig(
+  tokenProvider: TokenProvider,
+  inputs: Inputs,
+  siteName: string
+): Promise<[AzureUploadConfig, appService.WebSiteManagementClient]> {
+  // get publish credentials
+  const webSiteMgmtClient = await fetchWebSiteManagementClient(tokenProvider, inputs);
+  const listResponse = await AzureOperations.ListPublishingCredentials(
+    webSiteMgmtClient,
+    inputs.resourceGroupName,
+    siteName
+  );
+  const publishingUserName = listResponse.publishingUserName ? listResponse.publishingUserName : "";
+  const publishingPassword = listResponse.publishingPassword ? listResponse.publishingPassword : "";
+  const encryptedCredentials: string = utils.toBase64(
+    `${publishingUserName}:${publishingPassword}`
+  );
+  return [
+    {
+      headers: {
+        "Content-Type": "application/octet-stream",
+        "Cache-Control": "no-cache",
+        Authorization: `Basic ${encryptedCredentials}`,
+      },
+      maxContentLength: Infinity,
+      maxBodyLength: Infinity,
+      timeout: CommonConstants.deployTimeoutInMs,
+    },
+    webSiteMgmtClient,
+  ];
+}
+
+export async function azureWebSiteDeploy(
+  inputs: Inputs,
+  tokenProvider: TokenProvider,
+  buffer: Buffer,
+  siteName: string
+): Promise<appService.WebSiteManagementClient> {
+  const [config, client] = await getAzureDeployConfig(tokenProvider, inputs, siteName);
+  const zipDeployEndpoint: string = getZipDeployEndpoint(siteName);
+  const statusUrl = await AzureOperations.ZipDeployPackage(zipDeployEndpoint, buffer, config);
```

Comment:
The method name doesn't comply to TypeScript camel case convention, can we also refactor it later?